salt/states/firewalld.py

```diff
@@ -590,45 +590,46 @@ def _present(name,
                                            _current_port_fwd],
                                   'new': [fwd.todict() for fwd in port_fwd]}})
 
-    services = services or []
-    try:
-        _current_services = __salt__['firewalld.list_services'](name,
-            permanent=True)
-    except CommandExecutionError as err:
-        ret['comment'] = 'Error: {0}'.format(err)
-        return ret
+    if services or prune_services:
+      services = services or []
+      try:
+          _current_services = __salt__['firewalld.list_services'](name,
+              permanent=True)
+      except CommandExecutionError as err:
+          ret['comment'] = 'Error: {0}'.format(err)
+          return ret
 
-    new_services = set(services) - set(_current_services)
-    old_services = []
+      new_services = set(services) - set(_current_services)
```

Comment:
`services` would be set here. While it could be `None` after line 593, line 594 would catch that and set it to an empty list. An empty list will not throw a TypeError when cast to a set.

Your question made me wonder if `_current_services` could be `None` after the call to `firewalld.list_services` in line 596. I ran a quick test and verified that there was no error in the logs even when no services are allowed through the firewall. This indicates that the `firewalld.list_services` call will return an empty list, which would be ok when cast to a set in line 602.  Had it returned `None`, line 602 would have caused a TypeError.